### ISSUE TITLE
Add partition by file size

### ISSUE THREAD

**Issue by dziban303** [Tyrrrz/DiscordChatExporter#477]
### Tl;dr: add ability to partition output files by file size.

_You can really tell I wrote this at 4 in the morning, haha:_

Being able to partition large files at a specified file size would be a useful addition. Partitioning by message number is good, but depending on the content of the channel, file sizes of the output can vary significantly, and getting the "right" size can take a bit of guesswork. Messages in some channels—such as those used by a bot to post server events like user joins—are basically all the same size, but user off-topic channel message sizes are all over the place, with the result that one channel can produce files which are double the size of another channel. Setting a file size rather than a message count would take all the guesswork out of it, and produce similarly sized files in one batch without having to run the tool with different settings for different channels.

**Comment by andrewkolos** [comment]
@Tyrrrz If you are interested in this being supported, how would you like it work to work from the user's end? For starters, we could:

1) add a new CLI option, say `-pfs` or `--partitionByFileSize` or something like that or
2) reuse `-p` and have it work by file size instead of message count if the input describes a file size, for example `100mb` or `56kb`

**Comment by Tyrrrz** [comment]
I like the approach of reusing the same option. Probably would need to create some sort of abstraction like `IPartitioner` that has a method that takes the current message count and file size and makes the decision.

Also, before we start on this, we need to decide how the size limiting works. We won't know the output size ahead of time, so the only sensible approach I see is this (during export):

1. Flush to disk and measure the current file size
2. If file size below limit, continue
3. Repeat

If 2 fails, we stop enumerating messages and just finish early (but still finalize the document so that HTML and JSON aren't broken). This means we may likely end up with a file size which is higher than the defined limit, but probably not by much.

**Comment by andrewkolos** [comment]
Can't we just use the [Length property](https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.length?view=net-5.0#System_IO_Stream_Length) on `FileStream`?

**Comment by Tyrrrz** [comment]
@dziban303 is this issue still relevant to you?

**Comment by dziban303** [comment]
It will be as soon as I run another backup. Which I should've done a month ago. Thanks for the reminder, haha. Not sure why I didn't see the ping on #497, I will respond there. I will test this today and close if I don't run into trouble.

**Comment by dziban303** [comment]
OK, I tested the relevant CI build a few times with different partition sizes and it seems to work like a charm! When running my backup script it threw an error message that I don't think it's supposed to:

```
"A:\path\DiscordChatExporter.Cli.exe" export --channel a_channel --token a_token --output "A:\nother path\output" --format HtmlDark --dateformat "yyyy-MM-dd HH:mm:ss" --after "2021-01-13 00:10:00" --partition 10MB
Exporting channel 'A Channel'... ERROR: DiscordChatExporter.Domain.Exceptions.DiscordChatExporterException: No messages for the specified period.
  at DiscordChatExporter.Domain.Exporting.ChannelExporter.ExportChannelAsync(ExportRequest request, IProgress`1 progress) in
    D:\a\DiscordChatExporter\DiscordChatExporter\DiscordChatExporter.Domain\Exporting\ChannelExporter.cs:40
  at DiscordChatExporter.Domain.Exporting.ChannelExporter.ExportChannelAsync(ExportRequest request, IProgress`1 progress) in
    D:\a\DiscordChatExporter\DiscordChatExporter\DiscordChatExporter.Domain\Exporting\ChannelExporter.cs:62
  at DiscordChatExporter.Cli.Commands.Base.ExportCommandBase.ExportAsync(IConsole console, Guild guild, Channel channel) in
    D:\a\DiscordChatExporter\DiscordChatExporter\DiscordChatExporter.Cli\Commands\Base\ExportCommandBase.cs:62
  at DiscordChatExporter.Cli.Commands.Base.ExportCommandBase.ExportAsync(IConsole console, Channel channel) in
    D:\a\DiscordChatExporter\DiscordChatExporter\DiscordChatExporter.Cli\Commands\Base\ExportCommandBase.cs:69
  at DiscordChatExporter.Cli.Commands.Base.ExportCommandBase.ExportAsync(IConsole console, Snowflake channelId) in
    D:\a\DiscordChatExporter\DiscordChatExporter\DiscordChatExporter.Cli\Commands\Base\ExportCommandBase.cs:75
  at DiscordChatExporter.Cli.Commands.ExportChannelCommand.ExecuteAsync(IConsole console) in
    D:\a\DiscordChatExporter\DiscordChatExporter\DiscordChatExporter.Cli\Commands\ExportChannelCommand.cs:18
  at CliFx.CliApplication.RunAsync(IReadOnlyList`1 commandLineArguments, IReadOnlyDictionary`2 environmentVariables)
```

It's true, there weren't any messages for the time constraints. Later on in the script, it pulls a whole (small) guild, and there were some channels with no messages. Those got a normal message, e.g. `Channel 'eastern-gulf-coast': No messages for the specified period.` No biggie but possibly a scary message to some?

Anyway, yeah, it's doing exactly what I want. Thanks!
